lib/lrmd/lrmd_client.c

```diff
@@ -474,6 +474,56 @@ lrmd_tls_dispatch(gpointer userdata)
     return 1;
 }
 
+static int
+lrmd_tls_dispatch_async(gpointer userdata)
+{
+    lrmd_t *lrmd = userdata;
+    lrmd_private_t *native = lrmd->lrmd_private;
+    xmlNode *xml = NULL;
+    int rc = pcmk_rc_ok;
+
+    if (!remote_executor_connected(lrmd)) {
+        crm_trace("TLS dispatch triggered after disconnect");
+        return 0;
```

Comment:
I believe the only return values that matter for an fd dispatch function are -1 (to drop the connection) or anything else (to leave the connection up)

Reply:
I was hopeful this would fix my ctslab problems, but it doesn't look like it.  I'm now returning -1 in all cases except EAGAIN where I return 1.